client/packages/common/src/ui/forms/JsonForms/components/Text.tsx

```diff
@@ -11,23 +11,38 @@ export const stringTester = rankWith(3, schemaTypeIs('string'));
 
 const UIComponent = (props: ControlProps) => {
   const { data, handleChange, label, path, errors } = props;
-  const [localData, setLocalData] = useState<string>(data);
+  const [localData, setLocalData] = useState<string | undefined>(data);
+  // timestamp of the last key stroke
+  const [latestKey, setLatestKey] = useState<number>(0);
+  // debounce avoid rerendering the form on every key stroke which becomes a performance issue
   const onChange = useDebounceCallback(
     (value: string) => handleChange(path, value),
     [path]
   );
   const error = !!errors;
 
+  useEffect(() => {
```

Comment:
the intention of this appears to be the same as `useBufferState` - which is used to sync state and props. Would be less code to use that hook rather than reimplement, if it does what you need.

if not - I'd suggest making a hook out of this code so that it can be tested and reused

Reply:
Think useBufferState is different here since I have this last key stroke time check. The time check is needed cause otherwise it happens that text is changed inconsistently while typing. Maybe we can discuss in the office, maybe you have a better way to solve it as well since I still find it a bit hacky...